backend/clubs/views.py

```diff
@@ -547,7 +553,46 @@ def subscription(self, request, *args, **kwargs):
         including their names and emails.
         """
         club = self.get_object()
-        serializer = SubscribeSerializer(Subscribe.objects.filter(club=club), many=True)
+        if "date" in request.query_params.keys():
+            date = datetime.datetime.strptime(request.query_params["date"], "%Y-%m-%d")
+            serializer = SubscribeSerializer(
+                Subscribe.objects.filter(club=club, created_at__day=date.day), many=True
+            )
+        else:
+            serializer = SubscribeSerializer(Subscribe.objects.filter(club=club), many=True)
+
+        return Response(serializer.data)
+
+    @action(detail=True, methods=["get"])
+    def favorite(self, request, *args, **kwargs):
+        """
+        Return a list of all students that have favorited to the club.
+        """
+        club = self.get_object()
+        if "date" in request.query_params.keys():
+            date = datetime.datetime.strptime(request.query_params["date"], "%Y-%m-%d")
+            serializer = FavoriteTimeSerializer(
+                Favorite.objects.filter(club=club, created_at__day=date.day), many=True
+            )
+        else:
+            serializer = FavoriteTimeSerializer(Favorite.objects.filter(club=club), many=True)
+
+        return Response(serializer.data)
+
+    @action(detail=True, methods=["get"])
+    def club_visit(self, request, *args, **kwargs):
+        """
+        Return a list of all visits to a club page.
+        """
+        club = self.get_object()
+        if "date" in request.query_params.keys():
+            date = datetime.datetime.strptime(request.query_params["date"], "%Y-%m-%d")
+            serializer = ClubVisitSerializer(
+                ClubVisit.objects.filter(club=club, created_at__day=date.day), many=True
+            )
+        else:
+            serializer = ClubVisitSerializer(ClubVisit.objects.filter(club=club), many=True)
```

Comment:
We don't want clubs to see who visited their page individually, I think that's a little too much. Aggregate statistics with anonymous information should be good.